nemo/collections/asr/parts/submodules/ctc_decoding.py

```diff
@@ -607,6 +603,16 @@ def decode_ids_to_tokens(self, tokens: List[int]) -> List[str]:
         """
         raise NotImplementedError()
 
+    def decode_tokens_to_str_with_strip_punctuation(self, tokens: List[int]) -> str:
+        """
+        Decodes a list of tokens to a string and removes a space before supported punctuation marks.
+        """
+        text = self.decode_tokens_to_str(tokens)
+        if self.supported_punctuation:
+            punct_pattern = '|'.join([re.escape(p) for p in self.supported_punctuation])
```

Comment:
The `punct_pattern` can be constructed once in the constructor, and the regex can also be compiled once and stored in the constructor.